packages/react-native-bridge/common/gutenberg-web-single-block/content-functions.js

```diff
@@ -19,13 +25,56 @@ window.getHTMLPostContent = () => {
 };
 
 window.insertBlock = ( blockHTML ) => {
+	const {
+		blockEditorSelect,
+		blockEditorDispatch,
+	} = window.getBlockEditorStore();
+
 	// Setup the editor with the inserted block
 	const post = window.wp.data.select( 'core/editor' ).getCurrentPost();
 	window.wp.data
 		.dispatch( 'core/editor' )
 		.setupEditor( post, { content: blockHTML } );
 
 	// Select the first block
-	const clientId = window.blockEditorSelect.getBlocks()[ 0 ].clientId;
-	window.blockEditorDispatch.selectBlock( clientId );
+	const clientId = blockEditorSelect.getBlocks()[ 0 ].clientId;
+	blockEditorDispatch.selectBlock( clientId );
+
+	window.contentIncerted = true;
+};
+
+window.sendGutenbergReadyMessage = () => {
+	if ( window.webkit ) {
+		// iOS
+		window.webkit.messageHandlers.gutenbergReady.postMessage( '' );
+	} else {
+		// Android
+		window.wpwebkit.gutenbergReady();
+	}
+	window.readyMessageSent = true;
```

Comment:
On second thought, if `startObservingGutenberg` is called multiple times I suppose it may happen.
If so perhaps you could still avoid the extra `readyMessageSent` global variable by calling `unsubscribe()` here:
https://github.com/WordPress/gutenberg/pull/24658/files#diff-472501711ddd446bd17771beab9f6337R66

You would just need to declare `unsubscribe` beforehand so that it's accessible on that line.

Just a thought, not sure if it makes sense.

Reply:
So, you are right that the `wp.data.subscribe` callback is called many times very fast (basically on every redux action), and since `unsubscribe` _seems_ to be an async action, the callback can (and is) still called a few times after the un-subscription takes place.

`unsubscribe` is called as soon as we get an action that tells us the post is ready, and (again) is checked after every redux action.

In the other hand `window.startObservingGutenberg` is called once or just a few times, depending if it's WPCom, self-hosted, or Jetpack site, so calling `unsubscribe` outside the callback function probably won't have the desired effect.